src/main/java/com/mackervoy/calum/mud/content/ContentController.java

```diff
@@ -1,59 +1,69 @@
 /**
- * 
+ *
  */
 package com.mackervoy.calum.mud.content;
 
-import java.io.UnsupportedEncodingException;
-import java.net.URLDecoder;
-import java.nio.charset.StandardCharsets;
+import com.mackervoy.calum.mud.vocabularies.MUD;
+
+import java.io.ByteArrayOutputStream;
+import java.util.Optional;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.MediaType;
-
 import org.apache.jena.rdf.model.Model;
 import org.apache.jena.rdf.model.ModelFactory;
-import org.apache.jena.rdf.model.Property;
 import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
 import org.apache.jena.vocabulary.RDF;
 
-import com.mackervoy.calum.mud.MUDApplication;
-
 /**
  * @author Calum Mackervoy
- * Provides endpoints for client-side applications to request content 
+ * Provides endpoints for client-side applications to request content
  * (e.g. "please describe this object")
  */
 @Path("/content/")
 public class ContentController {
-	@GET
-    @Produces(MediaType.TEXT_PLAIN)
-    public String getContent(@QueryParam("uri") String uri) {
-		// bounds checking - could provide a catch-all describer this way
-		if(uri == null) uri = "";
-		
+  @GET
+  @Produces("text/turtle")
+  public String getContent(@QueryParam("uri") String uri) {
+		final String safeUri = uri == null ? "" : uri;
+		final Model m = ModelFactory.createDefaultModel();
+    m.read(safeUri);
+
+		return getDescriber(safeUri, m)
+			.map(describer -> describer.describe(m, safeUri))
+      .map(description -> descriptionToSensesModel(description))
```

Comment:
I think this stage of the pipeline should be managed by the `describer.describe` because  some describers might want to extend/replace the senses model, others not ?

Reply:
Ah good idea - so tweak the describer interace to return a sense model?

Reply:
Yeah I think so?

Reply:
I've made that change here: https://github.com/Multi-User-Domain/mud-jena/pull/7/commits/25f9538e247a2c2252c3bf3951527680b3abdaf2 😄

Reply:
Looks good!